charts/testkube-dashboard/templates/oauth2-deployment.yaml

```diff
@@ -56,52 +56,70 @@ spec:
         # Register a new application
         # https://github.com/settings/applications/new
         env:
+        {{- with .Values.oauth2.env }}
+        {{- if or .clientId .secretClientIdName }}
         - name: OAUTH2_PROXY_CLIENT_ID
-          {{- if .Values.oauth2.env.secretClientIdName }}
+          {{- if .secretClientIdName }}
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.oauth2.env.secretClientIdName }}
-              key: {{ .Values.oauth2.env.secretClientIdKey }}
+              name: {{ .secretClientIdName }}
+              key: {{ .secretClientIdKey }}
           {{- else }}
-          value: "{{ .Values.oauth2.env.clientId }}"
+          value: "{{ .clientId }}"
           {{- end }}
+        {{- end }}
+        {{- if or .clientSecret .secretClientSecretName }}
         - name: OAUTH2_PROXY_CLIENT_SECRET
-          {{- if .Values.oauth2.env.secretClientSecretName }}
+          {{- if .secretClientSecretName }}
           valueFrom:
             secretKeyRef:
               name: {{ .Values.oauth2.env.secretClientSecretName }}
```

Comment:
I guess this should be changed to `{{ .secretClientSecretName }}` and the key to `{{ .secretClientSecretKey }}`.